### ISSUE TITLE
Screen Reader: Many Pages: The fields of the code digits are announced without a name

### ISSUE THREAD

**Issue by olenamihno** [Expensify/App#77461]
## Action Performed:
1. Using Android TalkBack, open the Expensify app
2. Swipe to the code digits input field
3. Note how the field is announced

Other Occurrences:
1. Onboarding - Verify your email
2. On Profile - Verify Email
3. On Security - Merge Accounts
4. On Security - Merge Accounts - Enter Codecurrences:

## Expected Result:
Each field should be announced with its name, e.g.:
- 'verification code, digit 1 of 6'

## Actual Result:
The fields of the code digits are announced without a name.

## Area issue was found in:
Sign in

## Failed WCAG checkpoints
WCAG 4.1.2 Name, Role, Value
https://www.w3.org/WAI/WCAG22/Understanding/name-role-value

## User impact:
Providing role, state, and value information on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.

## Suggested resolution:
Use the edit field role and a descriptive accessibilityLabel, e..g.:
- 'verification code, digit 1 of 6'
https://reactnative.dev/docs/accessibility#accessibilitylabel

## Environment:
Samsung Galaxy S24 Ultra - Android 14, Partner - Native (No mobile browser)

## Screenshots/Videos

![Image](https://github.com/user-attachments/assets/b306b5a4-bb44-4a82-863f-d783831a52d8)

https://github.com/user-attachments/assets/275de77c-384b-44b7-ad18-e0e3feeb82de

**Version Number:** Production
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** No
**Logs:** https://stackoverflow.com/c/expensify/questions/4856  NA
**Notes/Photos/Videos:** Any additional supporting documentation  Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36628/community-issues/7006990
**Issue reported by:** NA
**Slack conversation:** NA
